app/controllers/user_notifications_controller.rb

```diff
@@ -0,0 +1,36 @@
+class UserNotificationsController < Api::V2::BaseController
+  before_action :require_login
+  before_action :find_resource, :only => [:update, :destroy]
+
+  def index
+    @notifications = NotificationRecipient.
+    where(:user_id => User.current.id).
+    order(:created_at).
+    eager_load(:notification, :notification_type)
+
+    render :json => {
+      :notifications => @notifications.paginate(paginate_options).map(&:payload),
+      :total => @notifications.count
+    }
+  end
+
+  def update
```

Comment:
I am missing "Mark all as read" action. IMHO it should be useful.
Other option would be bulk update action, It's much more efficient from DB point of view, and in the case of notifications IMHO we want to be as efficient as possible.